src/LibraryManager/LibraryConflictDetector.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager
+{
+    /// <summary>
+    /// Finds conflicts between different libraries, based on files brought in by each library.
+    /// </summary>
+    internal class LibraryConflictDetector
+    {
+        public LibraryConflictDetector(IDependencies dependencies, string defaultDestination, string defaultProvider)
+        {
+            _dependencies = dependencies ?? throw new ArgumentNullException(nameof(dependencies));
+            _defaultDestination = defaultDestination;
+            _defaultProvider = defaultProvider;
+            _fileToLibraryMap = new Dictionary<string, List<ILibraryInstallationState>>(RelativePathEqualityComparer.Instance);
+        }
+
+        private string _defaultDestination;
+        private string _defaultProvider;
+        private IDependencies _dependencies;
+        private Dictionary<string, List<ILibraryInstallationState>> _fileToLibraryMap;
+
+        public async Task<IEnumerable<FileConflict>> DetectConflictsAsync(
+            IEnumerable<ILibraryInstallationState> libraries,
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            _fileToLibraryMap.Clear();
```

Comment:
Can  _fileToLibraryMap be a local variable?

Reply:
It looks like it should be local, since this task is async which makes it more likely to end up with concurrent calls.